assets/js/modules/analytics/setup.js

```diff
@@ -368,21 +331,37 @@ class AnalyticsSetup extends Component {
 				} );
 			}
 
-			return new Promise( ( resolve ) => {
-				resolve( newState );
-			} );
+			// If tag hasn't been detected in wp_head, look for existing tag in the html.
+			if ( ! newState.existingTag && ! newState.errorCode ) {
+				const existingTag = await getExistingTag( 'analytics' );
+
+				if ( existingTag && existingTag.length  ) {
+
+					// Verify the user has access to existing tag if found. If no access request will return 403 error and catch err.
+					await data.get( 'modules', 'analytics', 'tag-permission', { tag: existingTag }, false );
+					newState = Object.assign( newState, {
+						existingTag,
+						useSnippet: ( ! existingTag && ! onSettingsPage ) ? true : useSnippet,
```

Comment:
The condition will never evaluate to true there because `! existingTag` is never true, see the wrapping condition. We can just pass `useSnippet`.